modules/programs/qutebrowser.nix

```
@@ -274,13 +291,26 @@ in {
       ++ optional (!cfg.enableDefaultBindings) "c.bindings.default = {}"
       ++ mapAttrsToList formatKeyBindings cfg.keyBindings
       ++ optional (cfg.extraConfig != "") cfg.extraConfig);
+
+    quickmarksFile = optionals (cfg.quickmarks != { }) concatStringsSep "\n"
+      ((mapAttrsToList formatQuickmarks cfg.quickmarks));
```

Hmm, isn't the `(` in the wrong place? I would have expected
```suggestion
    quickmarksFile = optionals (cfg.quickmarks != { }) (concatStringsSep "\n"
      (mapAttrsToList formatQuickmarks cfg.quickmarks));
```

---

Me as well, and that's how I initially wrote it. The `format` utility converted it to this, so i assumed that is prefered.

---

Oh, `optionals` doesn't check whether the last argument is actually a list so it returns whatever is on the right, in this case the function `concatStringsSep`, which is then invoked with the rest.

https://github.com/NixOS/nixpkgs/blob/189a13688782682f2022023c025733bfb269f9d3/lib/lists.nix#L266-L270
``` nix
{
  optionals =
    # Condition
    cond:
    # List to return if condition is true
    elems: if cond then elems else [];
}
```